Homework2/MatrixTask/Extension.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MatrixTask
+{
+    public delegate T Add<T>(T elementOne, T elementTwo);
+
+    public static class Extension
+        {
+
+        public static T MatrixSum<T>(this Matrix<T> matrixOne,int firstElementPosition, Matrix<T> matrixTwo, int secondElementPosition, Add<T> delegates)
+        {
+            return delegates(matrixOne[firstElementPosition, firstElementPosition], matrixTwo[secondElementPosition, secondElementPosition]);
```

Comment:
Nope .
here is the logic to follow.
Also it is not a complete flow, it covers only the case when matrixOne size is the same as matrixTwo size. 

```suggestion
            public static T MatrixSum<T>(this Matrix<T> matrixOne,Matrix<T> matrixTwo, Add<T> addition)
        {
              var newMatrix = new Matrix(matrixOne.Size);
               
               //The code assumes that matrixOne and matrixTwo sizes are equal. 
              for(int i = 0; i < matrixOne.Size; i++)
              {
                 newMatrix[i,i] = addition(  matrixOne[i,i] , matrixTwo[i,i]);
              }
              
```